Migration to AsciiDoc (#549)
OpenSimulationInterface/open-simulation-interface: doc/architecture/sensor_view_configuration.adoc

= Sensor view configuration

The sensor view configuration configures the sensor view.
There are two sources for sensor-view-configuration data.

> Sensor-view-configuration data can be provided by the environment simulation.

Reviewer: ```suggestion
Sensor-view-configuration data can be provided by the simulation environment.
```
Or to the "environment simulator"

Reviewer 2: CCB Meeting 04.08.2021:
- rejected as per discussion.

> In this case, the data is intended for the automatic configuration of an environment simulation in order to supply the necessary input to a sensor model.

Reviewer: ```suggestion
In this case, the data is intended to supply the necessary input to a sensor model.
```

> Sensor-view-configuration data is not intended to provide parameters to a generic sensor model. 

Reviewer: ```suggestion
Sensor-view-configuration data is not intended to provide exact parameters to a sensor model. 
```

Reviewer 2: CCB Meeting 04.08.2021:
- Sensor-view-configuration data is not intended to provide parameters to a sensor model.

> Sensor-view-configuration data can also be provided by the sensor model to the environment simulation.

Reviewer: ```suggestion
Sensor-view-configuration data can also be provided by the sensor model to the simulation environment.
```
same as above or "environment simulator"

Author: No change, because we agreed in CCB Meeting, 04.08.2021 that "environment simulation" is correct.

> In this case, the data describes the input configuration that is desired by the sensor model.

Reviewer: ```suggestion
In this case, the data describes the configuration that is requested by the sensor model.
```

> In response, the environment simulation will configure the input and provide a new message which describes the actual configuration.

Reviewer: ```suggestion
In response, the simulation environment will reply with a new sensor-view-configuration message that describes the possible configuration.
```

Reviewer 2: CCB Meeting 04.08.2021:
- wording "environment simulation" should be defined somewhere. --> Part that simulates the environement.
- "simulation environment" is not the correct wording here. "environment simulation" should be used here.

Author: @kmeids : Apart from "environment simulation" vs "simulation environment", your suggestion also contains some major changes:

- The environment does not configure the input.
- The message does not describe the actual configuration but possible configurations.

 Should I still apply these changes? Or is this moot since we talk about  "environment simulations" and not "simulation environment"? I hope I am not confusing anything here.

Author: @stefancyliax
- The word "environment simulation" is mentioned here: https://github.com/OpenSimulationInterface/open-simulation-interface/blob/migration/doc/images/osi-context.png.
- This image is used here: https://github.com/OpenSimulationInterface/open-simulation-interface/blob/migration/doc/architecture/architecture_overview.adoc

I recommend to add a sentence in this topic. What do you think?

Reviewer 3: CCB: @pmai will reorder sentences and do slight rewrite to address these comments.

Reviewer 2: @pmai is this done?

Reviewer 2: OSI CCB: 
- @pasched Please review the new chapter from pmai.

> The configuration requested by the sensor model may differ from the actual configuration.

Reviewer: ```suggestion
The configuration requested by the sensor model may differ from the one provided by the simulation environment.
```

> This happens when the environment simulation does not support a given requested configuration or when the requested configuration allowed is ambiguous.

Reviewer: ```suggestion
This happens when the simulation environment does not support a requested configuration or when the requested configuration is ambiguous.
```

> The message describing the requested configuration differs from the message describing the actual configuration.

Reviewer: ```suggestion
The message describing the requested (by the sensor model) configuration differs from the message describing the actual configuration (provided by the simulation environment).
```

= Sensor view configuration

The sensor view configuration configures the sensor view.
Sensor-view-configuration data enables the environment simulation to supply the necessary input to a sensor model.

There are two sources for sensor-view-configuration data.

Sensor-view-configuration data can be provided by the environment simulation.
Note that the data is intended for the automatic configuration of an environment simulation.

> Sensor-view-configuration data is not intended to provide parameters to a sensor model. 

Reviewer 4: What is the intention then? Shouldn't it be included then directly in SensorView?

= Sensor view configuration

The sensor view is flexibly defined to support different kinds of sensor models with appropriate input.
The sensor view configuration defines the configuration of a particular sensor view.

> This message is used in the initialization phase of a simulation to negotiate the sensor view configuration for a particular sensor view input.

Reviewer 5: @pmai: What do you mean with "This message"? 
Is it the "sensor-view-configuration message"?
